### PR TITLE
Added docs on JSON/YAML schema

### PR DESCRIPTION
**Pull request by jilleJr**
> For Iver employees: WO0000000681896

Was previously hosted at https://gist.github.com/jilleJr/07bcb76f86bcca3a398d42a51b3d303f

No need now that we have a public documentations page.

The `.json` file path is kind of a guess, but as the images from `_images` works that way so it _should_ work.

## Preview

> https://github.com/iver-wharf/iver-wharf.github.io/blob/27e43d58db551c7b86a8bb1d1ba3a3aa40da11bd/docs/usage-wharfyml/json-yaml-schema-for-wharf-ci-yml.md
>
> image

### CODE REVIEW

File: docs/_static/wharf-ci-schema.json [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+                        "certificatesMountPath": {
+                            "type": "string",
+                            "examples": [
+                                "/usr/local/share/ca-certificates"
+                            ]
+                        },
+                        "secretName": {
+                            "type": "string",
+                            "examples": [
+                                "spark-system-tests-secret"
```
Comment (iverestefans) [inline docs/_static/wharf-ci-schema.json:280]: Do we want to publish it?

Comment (iverestefans) [inline docs/_static/wharf-ci-schema.json:280]: Do we want to publish it?

File: docs/_static/wharf-ci-schema.json [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+                            ]
+                        },
+                        "force": {
+                            "type": "boolean",
+                            "default": false
+                        },
+                        "cluster": {
+                            "type": "string",
+                            "examples": [
+                                "kubectl-config",
```
Comment (iverestefans) [inline docs/_static/wharf-ci-schema.json:334]: Maybe here should be different examples.

File: docs/_static/wharf-ci-schema.json [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+                                "v2.14.1",
+                                "v3.0.2",
+                                "v3.2.0",
+                                "latest"
+                            ]
+                        },
+                        "cluster": {
+                            "type": "string",
+                            "examples": [
+                                "kubectl-config",
```
Comment (iverestefans) [inline docs/_static/wharf-ci-schema.json:418]: Maybe there should be different examples.

File: docs/_static/wharf-ci-schema.json [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+                            "examples": [
+                                "src/api"
+                            ]
+                        },
+                        "secret": {
+                            "type": "string"
+                        },
+                        "registry": {
+                            "type": "string",
+                            "default": "https://harbor.dgc.local"
```
[the remark this answers is not in the document]
Comment (jilleJr) [reply docs/_static/wharf-ci-schema.json:207]: Hmm good point. I didn't actually sanitize this file. I'll just remove the defaults and add an "example" tag instead with just `https://harbor.local`

File: docs/_static/wharf-ci-schema.json (hunk 1 above)
Comment (jilleJr) [reply docs/_static/wharf-ci-schema.json:280]: Changed to `myproject-system-tests-secret`

File: docs/_static/wharf-ci-schema.json (hunk 2 above)
Comment (jilleJr) [reply docs/_static/wharf-ci-schema.json]: Changed to `stage-cluster-config` and `prod-cluster-config` to mimic #1

File: docs/_static/wharf-ci-schema.json (hunk 3 above)
Comment (jilleJr) [reply docs/_static/wharf-ci-schema.json]: Changed to `stage-cluster-config` and `prod-cluster-config` to mimic #1

File: docs/_static/wharf-ci-schema.json [hunk 5]
```diff
@@ -0,0 +1,501 @@
...
+                        "version": {
+                            "type": "string",
+                            "examples": [
+                                "${GIT_TAG}"
+                            ]
+                        },
+                        "repo": {
+                            "type": "string",
+                            "examples": [
+                                "http://nuget.dgc.local/libs/api/v2/package"
```
[the remark this answers is not in the document]
Comment (jilleJr) [reply docs/_static/wharf-ci-schema.json]: Changed to `nuget.local`

[the remark this answers is not in the document]
Comment (jilleJr) [reply docs/_static/wharf-ci-schema.json]: Changed to `nuget.local`

File: docs/_static/wharf-ci-schema.json [hunk 6]
```diff
@@ -0,0 +1,501 @@
...
+                    "required": [
+                        "project-path",
+                        "version",
+                        "repo"
+                    ],
+                    "properties": {
+                        "project-path": {
+                            "type": "string",
+                            "examples": [
+                                "src/Iver.Spark.Service.Api.Client"
```
[the remark this answers is not in the document]
Comment (jilleJr) [reply docs/_static/wharf-ci-schema.json]: Changed to `src/MyProject` to mimic #1

File: docs/_static/wharf-ci-schema.json [hunk 7]
```diff
@@ -0,0 +1,501 @@
...
+                        },
+                        "chart-path": {
+                            "type": "string",
+                            "examples": [
+                                "/web-app"
+                            ]
+                        },
+                        "destination": {
+                            "type": "string",
+                            "default": "https://harbor.dgc.local/chartrepo/default"
```
[the remark this answers is not in the document]
Comment (jilleJr) [reply docs/_static/wharf-ci-schema.json]: Changed to `harbor.local`

File: docs/_static/wharf-ci-schema.json [hunk 8]
```diff
@@ -0,0 +1,501 @@
...
+                        },
+                        "namespace": {
+                            "type": "string",
+                            "examples": [
+                                "stage"
+                            ]
+                        },
+                        "repo": {
+                            "type": "string",
+                            "default": "https://harbor.dgc.local/chartrepo"
```
[the remark this answers is not in the document]
Comment (jilleJr) [reply docs/_static/wharf-ci-schema.json]: Changed to `harbor.local`

File: docs/_static/wharf-ci-schema.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (jilleJr) [reply docs/_static/wharf-ci-schema.json]: Changed to `harbor.local`

File: docs/usage-wharfyml/json-yaml-schema-for-wharf-ci-yml.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+[The language server from Red Hat](https://github.com/redhat-developer/yaml-language-server#readme)
+that most editors use to have YAML syntax and intellisense, has support for JSON
+schemas.
+
+Wharf has such a schema for their `.wharf-ci.yml` file. 
+
+## The JSON schema
+
+File is hosted here: <https://iver-wharf.gitlab.io/_static/wharf-ci-schema.json>
```
[the remark this answers is not in the document]
Comment (jilleJr) [reply docs/usage-wharfyml/json-yaml-schema-for-wharf-ci-yml.md]: Fixed link. Typo, meant to be iver-wharf.gitHUB.io, not iver-wharf.gitLAB.io

**Review by iverestefans — APPROVED** [review]
LGTM!

**Review by vreve — APPROVED** [review]

**Pull request merged**